core/src/main/scala/filodb.core/query/RangeVector.scala

```diff
@@ -167,22 +167,31 @@ object SerializableRangeVector extends StrictLogging {
    * The most efficient option when you need to create multiple SRVs as the containers are automatically
    * shared correctly.
    * The containers are sent whole as most likely more than one would be sent, so they should mostly be packed.
+   * limit - to limit the rows added to the result
    */
   def apply(rv: RangeVector,
             builder: RecordBuilder,
-            schema: RecordSchema): SerializableRangeVector = {
+            schema: RecordSchema,
+            resultSize: Int = 0,
+            limit: Int = Int.MaxValue,
+            failOnLimit: Boolean = true): SerializableRangeVector = {
```

Comment:
Why does failOnLimit default to true?  From the description of the PR, it sounds like we don't want to fail on limit, just to truncate.